packages/core/src/transaction.ts

```diff
@@ -159,20 +171,16 @@ export class Transaction {
 
     this.#connection = connection;
 
-    let result;
     try {
       await this.begin();
-
-      result = await this.setDeferrable();
+      await this.setDeferrable();
```

Comment:
`setDeferrable`, `begin`, `cleanup`, and `forceCleanup` should be `#private`

Similarly, if `finished` is not used externally anymore, should be `#private`

Reply:
Finished is being used by the following function inside the `queryRaw` function.

https://github.com/sequelize/sequelize/blob/23cdf1de2dc76aa0f063719045bae25162856493/packages/core/src/sequelize.js#L706-L712

Reply:
I've set finished to private and added a getter to be able to access the value outside the class.